Boyer-Moore-Horspool/BoyerMooreHorspool.swift

```diff
@@ -6,70 +6,78 @@
   http://www.drdobbs.com/database/faster-string-searches/184408171
 */
 extension String {
-    func index(of pattern: String, usingHorspoolImprovement: Bool = false) -> Index? {
-        // Cache the length of the search pattern because we're going to
-        // use it a few times and it's expensive to calculate.
-        let patternLength = pattern.count
-        guard patternLength > 0, patternLength <= self.count else { return nil }
-      
-        // Make the skip table. This table determines how far we skip ahead
-        // when a character from the pattern is found.
-        var skipTable = [Character: Int]()
-        for (i, c) in pattern.enumerated() {
-            skipTable[c] = patternLength - i - 1
-        }
-
-        // This points at the last character in the pattern.
-        let p = pattern.index(before: pattern.endIndex)
-        let lastChar = pattern[p]
-
-        // The pattern is scanned right-to-left, so skip ahead in the string by
-        // the length of the pattern. (Minus 1 because startIndex already points
-        // at the first character in the source string.)
-        var i = index(startIndex, offsetBy: patternLength - 1)
-
-        // This is a helper function that steps backwards through both strings
-        // until we find a character that doesn’t match, or until we’ve reached
-        // the beginning of the pattern.
-        func backwards() -> Index? {
-            var q = p
-            var j = i
-            while q > pattern.startIndex {
-                j = index(before: j)
-                q = index(before: q)
-                if self[j] != pattern[q] { return nil }
-            }
-            return j
+  func index(of pattern: String, usingHorspoolImprovement: Bool = false) -> Index? {
```

Comment:
Indentation was changed from 4 spaces to 2 spaces, hiding any changes in this commit diff.

Style guide does say to use 2 spaces...

 > https://github.com/raywenderlich/swift-style-guide#spacing

But I would suggest fixing the indenting in a separate commit and/or PR, then making the algorithm changes in the next.

Reply:
Now it's okay? Or maybe create one more PR?

Reply:
That does now make the "Files Changed" diff much easier to read.

Personally I would re-create and re-push this branch to remove those indentation commits entirely from the git history, but others may have different opinions there. Not sure what the repo's standards are.

Reply:
Thanks for pointing this out @pkamb